vta/hardware/chisel/src/main/scala/shell/SimShell.scala

```diff
@@ -61,18 +62,33 @@ class VTAMem(implicit p: Parameters) extends Module {
   mem_axi.io.axi <> io.axi
 }
 
+/** VTASim.
+  *
+  * This module is used to handle simulation, including wait, resume, and exit.
+  */
+class VTASim(implicit p: Parameters) extends MultiIOModule {
+  val sim_wait = IO(Output(Bool()))
+  val sim = Module(new VTASimDPI)
+  sim.io.reset := reset
+  sim.io.clock := clock
+  sim_wait := sim.io.dpi_wait
+}
 /** SimShell.
   *
   * The simulation shell instantiate a host and memory simulation modules and it is
   * intended to be connected to the VTAShell.
   */
```

Comment:
Can we add more details on the shell interface spec? For instance, what these new signals sim_clock and sim_wait are used for? And how the mod_sim module interacts with the sim shell. I think that extra comments here and above won't hurt.

Reply:
Great catch, just did.